### PR TITLE
Vitis-5925: Improve bandwidth calculation in profiling summary

### PR DESCRIPTION
**Pull request by jvillarre**
#### Problem solved by the commit
When enabled, the profile summary reports kernel to global memory data transfers, and gives an average bandwidth utilization.  This bandwidth utilization was calculated using assumptions on maximum bandwidth between the kernel and memory that did not take into account differences of connections, such as bit-width and clock rate.

This pull request improves the kernel to global memory table by reporting the achieved bandwidth that was observed as well as the ideal bandwidth on an ideal connection.  The ideal bandwidth is based on the maximum bit-width of a connection and the maximum clock speed of that connection, while the achieved bandwidth is based on the actual connection's bit-width, actual connection's clock rate, and the memory it is connected to.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
Incorrect or misleading values in the profile summary tables were discovered by manual inspection and comparison with expected values.

#### How problem was solved, alternative solutions (if any) and why they were rejected
The profiling library now parses additional information in the system metadata section in the xclbin to determine the characteristics of each port of a compute unit.  This includes bit-width, clock speed, and the memory connection.  This information is now collected as a new Port struct which is stored as part of a compute unit struct in the profiling side.

This implementation had the additional benefit of collecting the arguments associated with each port which was previously only available in OpenCL flows, so this information is now available in Native XRT flows as well.

#### Risks (if any) associated the changes in the commit
Low risk as the bandwidth numbers previously reported were unclear, and if the metadata is not available then numbers will not be reported.

#### What has been tested and how, request additional testing if necessary
The changes have been run on various different hardware configurations of ports in order to see that the calculations are working as expected.  Full testing on large regression suites will be necessary to test platform specific information.

#### Documentation impact (if any)
Documentation on the profiling table columns will have to be updated to reflect the new fields and what they mean.

### CODE REVIEW

**Review by pgschuey — APPROVED** [review]

File: src/runtime_src/xdp/profile/database/static_info_database.cpp [hunk 1]
```diff
@@ -1428,57 +1403,213 @@ namespace xdp {
...
+        for (auto& kernel : region.second.get_child("kernels")) {
+          auto kernelName = kernel.second.get<std::string>("name", "");
+          for (auto& port : kernel.second.get_child("ports")) {
+            auto portName = port.second.get<std::string>("name");
+            auto portType = port.second.get<std::string>("port_type");
+            if (portName == "S_AXI_CONTROL" || portType == "stream")
+              continue;
+
+            auto portWidth = port.second.get<std::string>("data_width");
+	    std::transform(portName.begin(), portName.end(), portName.begin(),
```
Comment (hackwa) [inline src/runtime_src/xdp/profile/database/static_info_database.cpp:1517]: looks misaligned. Mix of tabs and spaces?

Comment (jvillarre) [reply src/runtime_src/xdp/profile/database/static_info_database.cpp]: Changed these lines to spaces to align in the latest commit

**Review by hackwa — APPROVED** [review]
looks good!

**Comment by dayeh-xilinx** [comment]
retest this please.

**Comment by dayeh-xilinx** [comment]
retest this please.

**Pull request merged**